tango_simlib/tests/test_tango_sim_generator.py

```diff
@@ -154,3 +162,122 @@ def test_sim_control_device_attribute_change(self):
         self.sim_control_device.pause_active = True
         setattr(self.sim_control_device, 'last_val', input_value)
         self.assertEqual(self.sim_device.temperature, input_value)
+
+class test_TangoSimGenDeviceIntegrationForFandangoFile(ClassCleanupUnittestMixin,
+    unittest.TestCase):
+
+    longMessage = True
+
+    @classmethod
+    def setUpClassWithCleanup(cls):
+        cls.port = helper_module.get_port()
+        cls.host = helper_module.get_host_address()
+        cls.data_descr_file = [pkg_resources.resource_filename(
+            'tango_simlib.tests.config_files', 'database2.fgo')]
+        cls.temp_dir = tempfile.mkdtemp()
+        cls.sim_device_class = tango_sim_generator.get_device_class(cls.data_descr_file)
+        device_name = 'test/nodb/tangodeviceserver'
+        server_name = 'databaseds'
+        server_instance = 'test'
+        database_filename = '%s/%s_tango.db' % (cls.temp_dir, server_name)
+        sim_test_device_prop = dict(model_key=device_name)
+        tango_sim_generator.generate_device_server(
+                server_name, cls.data_descr_file, cls.temp_dir)
+        helper_module.append_device_to_db_file(
+                server_name, server_instance, device_name,
+                database_filename, cls.sim_device_class)
+        cls.db_instance = helper_module.append_device_to_db_file(
+                            server_name, server_instance, '%scontrol' % device_name,
+                            database_filename, '%sSimControl' % cls.sim_device_class,
+                            sim_test_device_prop)
+        cls.sub_proc = subprocess.Popen(
+            ["python", "{}/{}".format(cls.temp_dir, server_name),
+             server_instance, "-file={}".format(database_filename),
+             "-ORBendPoint", "giop:tcp::{}".format(cls.port)])
+        # Note that tango demands that connection to the server must
+        # be delayed by atleast 1000 ms of device server start up.
+        time.sleep(1)
+        cls.sim_device = tango.DeviceProxy(
+                '%s:%s/test/nodb/tangodeviceserver#dbase=no' % (
+                    cls.host, cls.port))
+        cls.sim_control_device = tango.DeviceProxy(
+                '%s:%s/test/nodb/tangodeviceservercontrol#dbase=no' % (
+                    cls.host, cls.port))
+        cls.addCleanupClass(cls.sub_proc.kill)
+        cls.addCleanupClass(shutil.rmtree, cls.temp_dir)
```

Comment:
Agree with @kmadisa that there is a lot of duplication now.  Would be nice to share some of the code with tests for all parsers.  Also notice that the simdd format doesn't get tested here.  Suggest we leave the rework into a base class for another day, and another PR.

Reply:
ok